Assets/HoloToolkit/Input/Scripts/GazeManager.cs

```diff
@@ -151,10 +164,34 @@ private void UpdateRaycast()
             {
                 if (oldFocusedObject != null)
                 {
+                    IInteractable oldInteractable = oldFocusedObject.GetComponent<IInteractable>();
+                    if (oldInteractable != null)
+                    {
+                        oldInteractable.OnGazeExit();
+                    }
+
+                    if (OnGazeExit != null)
+                    {
+                        OnGazeExit(oldFocusedObject);
+                    }
+
+                    // Obsolete! Please subscribe to OnGazeExit event.
                     oldFocusedObject.SendMessage("OnGazeLeave", SendMessageOptions.DontRequireReceiver);
                 }
                 if (FocusedObject != null)
                 {
+                    IInteractable newInteractable = FocusedObject.GetComponent<IInteractable>();
+                    if (newInteractable != null)
+                    {
+                        newInteractable.OnGazeEnter();
+                    }
+
+                    if (OnGazeEnter != null)
```

Comment:
Do we like the fact that we are supporting 3 different ways of doing this? Feels confusing to me.
Perhaps the default GazeManager supports only one way and other classes derived from partial support the other mechanisms.